File: src/main/java/com/mrpowergamerbr/loritta/utils/Jankenpon.java
package com.mrpowergamerbr.loritta.utils;

import lombok.Getter;

@Getter
public enum Jankenpon {
	ROCK("pedra", ":black_circle:", "SCISSORS", "PAPER"),
	PAPER("papel", ":newspaper:", "ROCK", "SCISSORS"),
	SCISSORS("tesoura", ":scissors:", "PAPER", "ROCK");
	
	String lang;
	String emoji;
	String wins;
	String loses;
	
	Jankenpon(String lang, String emoji, String wins, String loses) {
		this.lang = lang;
		this.emoji = emoji;
		this.wins = wins;
		this.loses = loses;
	}
	
	public JankenponStatus getStatus(Jankenpon janken) {
		if (this.name().equalsIgnoreCase(janken.loses)) {
			return JankenponStatus.WIN;
		}
		if (this == janken) {
			return JankenponStatus.DRAW;
		}
		return JankenponStatus.LOSE;
	}
	
	public static Jankenpon getFromLangString(String str) {
		for (Jankenpon janken : Jankenpon.values()) {
			if (janken.lang.equals(str)) {
				return janken;
			}
		}
		return null;
	}
	public enum JankenponStatus {
		WIN,
		LOSE,
		DRAW;
	}
}

File: src/main/java/com/mrpowergamerbr/loritta/utils/Jankenpon.kt
package com.mrpowergamerbr.loritta.utils

import com.mrpowergamerbr.loritta.utils.locale.BaseLocale

enum class Jankenpon(var lang: String, var emoji: String, var wins: String, var loses: String) {
	// Os wins e os loses precisam ser uma string já que os enums ainda não foram inicializados
	ROCK("PPT_Rock", "\uD83C\uDF11", "SCISSORS", "PAPER"),
	PAPER("PPT_Paper", ":newspaper:", "ROCK", "SCISSORS"),
	SCISSORS("PPT_Scissors", ":scissors:", "PAPER", "ROCK");

	fun getStatus(janken: Jankenpon): JankenponStatus {
		if (this.name.equals(janken.loses, ignoreCase = true)) {
			return JankenponStatus.WIN
		}
		if (this == janken) {
			return JankenponStatus.DRAW
		}
		return JankenponStatus.LOSE
	}

	enum class JankenponStatus {
		WIN,
		LOSE,
		DRAW
	}

	companion object {
		fun getFromLangString(str: String, locale: BaseLocale): Jankenpon? {
			for (janken in Jankenpon.values()) {
				if (str == locale[janken.lang]) {
					return janken
				}
			}
			return null
		}
	}
}